src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1452,13 +1456,8 @@ ReadOnlyTask(TaskType taskType)
  * contains a FOR UPDATE clause that locks any reference tables.
  */
 static bool
-SelectForUpdateOnReferenceTable(RowModifyLevel modLevel, List *taskList)
+SelectForUpdateOnReferenceTable(List *taskList)
 {
-	if (modLevel != ROW_MODIFY_READONLY)
```

Comment:
What do you think about adding `Assert(modeLevel != ROW_MODIFY_READONLY)` since it seems to be expected?

Reply:
Reasonable, will do

Reply:
Thought more on this. I disagree. The `ROW_MODIFY_READONLY` is from the assumption that it's equivalent to checking `commandType == CMD_SELECT`. But that won't hold later if one has a modifying CTE alongside a `SELECT FOR UPDATE`

It could also happen that `modLevel` is `ROW_MODIFY_NONE`, but in that case I expect this function still returns false because such tasks shouldn't have any relationRowLockList elements